subworkflows/local/consensus/main.nf

```diff
@@ -11,63 +11,76 @@ include { FGBIO_ZIPPERBAMS                  } from '../../../modules/nf-core/fgb
 include { SAMTOOLS_FASTQ                    } from '../../../modules/nf-core/samtools/fastq/main'
 include { SAMTOOLS_INDEX                    } from '../../../modules/nf-core/samtools/index/main'
 include { BWA_MEM                        } from '../../../modules/nf-core/bwa/mem/main'
-include { samplesheetToList } from 'plugin/nf-schema'
 
 
 workflow CONSENSUS {
     take:
         ch_input_fastq                   // channel: [meta_with_readgroup, fastq] for SE/PE/duplex samples
         ch_genomes                       // map: reference genome files
-        ch_umi_in_readname               // boolean
-    main:
 
-        ch_versions       = Channel.empty()
+    main:
+        def ch_versions       = Channel.empty()
+        def readname_fastq     = Channel.empty()
+        def ch_readname_uBAM   = Channel.empty()
+        def ch_fastqtobam_with_bai = Channel.empty()
 
         ch_input_fastq
-            .combine(ch_umi_in_readname)
+            .combine(params.umi_in_readname)
             .map { meta, fq1, fq2, umi_flag ->
                 tuple(meta, fq1, fq2, umi_flag)
             }
-            .branch { _meta, _fq1, _fq2, umi_flag ->
-                umi_in_readname: umi_flag == true
-                umi_in_seq: umi_flag == false
-            }
-            .set  {ch_input_fastq_branch}
+            .set  {ch_input_fastq_combined}
 
 
-    // Readname branch
+        if (params.umi_in_readname) {
 
-        // 1.1: FASTQ => uBAM
-        ch_RN_uBAM = Channel.empty()
-        RN_FQ  = ch_input_fastq_branch.umi_in_readname.map { meta, r1, r2, _f -> tuple(meta, [r1, r2]) }
+        // Case 1: UMI_in_readname
 
-        FASTQTOBAM_READNAME(RN_FQ)
-        SAMTOOLS_INDEX(FASTQTOBAM_READNAME.out.bam)
+            ch_input_fastq_combined
+                .filter { _meta, _fq1, _fq2, umi_flag -> umi_flag == true }
```

Comment:
No need to filter in this case, either all entries are `true` or all entries are `false` depending on the input parameter